### PR TITLE
[RNMobile] Fix crash bringing gutenberg master to mobile

### PR DESCRIPTION
**Pull request by etoledom**
## Description

This patch makes the minimal adjustments to make the native version of gutenberg work with the latest change to the core data entities. Since we don't yet have networking support on mobile, the approach explored here is to preload the store with the required core data entities so it does not fail on load.

### Testing Instruction

- Checkout Gutenberg Mobile PR https://github.com/wordpress-mobile/gutenberg-mobile/pull/1366 and update submodule
- Start demo app `yarn start:reset` and `yarn android`
- Check that the app boots and that nothing is broken in the UI
- Check that the tests pass

### CODE REVIEW

**Comment by gziolo** [comment]
I see many changes which were reviewed in the past and merged into `rnmobile/master`. Is it possible to identify only those changes which are necessary to merge `rnmobile/master` into `master`? This would make the whole process much easier.

File: packages/rich-text/src/component/index.native.js
Comment (gziolo) [inline packages/rich-text/src/component/index.native.js]: That's interesting. It's not part of the web public API. Do you think it would be helpful to start using the wrapper components for theming in general?

File: packages/rich-text/src/component/index.native.js
Comment (gziolo) [inline packages/rich-text/src/component/index.native.js]: I don't think it's a good idea to prefix the prop name with `use` as this confuses it with React hooks.

File: packages/block-library/src/separator/edit.js
Comment (gziolo) [inline packages/block-library/src/separator/edit.js]: The following might be easier to follow for those not familiar with the internal implementation:
```suggestion
			<SeparatorInspectorConrols
```

File: .travis.yml
Comment (gziolo) [inline .travis.yml]: Yes, let's do it if you plan to use `rnmobile/master` more often 👍

**Review by gziolo — COMMENTED** [review]
General observation, I noticed that you introduce new public APIs which are used only in native implementations. `withTheme` is something introduced recently. I know that there is `mobile` subfolder in `@wordpress/components`. How about, we always prefix them with either `__unstable` or `__experimental` to make sure they are never documented until we consider them as production-ready. I have a feeling that the fact that you introduced them doesn't mean they fit mobile use case only. I can see `withTheme` applicable to both web and mobile. Bottom sheet component is another example where the pattern used for native mobile would fit the web mobile as well. Having this clear indicator that some new APIs were created would help to identify them and think more broadly how to expose them for web.

/cc @hypest @koke @pinarol @Tug

File: .travis.yml
Comment (Tug) [reply .travis.yml]: We might want to revert that I'm not sure. I merely merged `rnmobile/master` in here (thus the amount of changes unrelated to the PR description) but it might need some work before we merge it back to master

File: packages/rich-text/src/component/index.native.js
Comment (Tug) [reply packages/rich-text/src/component/index.native.js]: Agreed here. cc @etoledom

Comment (Tug) [reply packages/rich-text/src/component/index.native.js]: Something like `extendStyles()` or `extendStylesToTheme()` maybe?

**Comment by pinarol** [comment]
I agree about marking `withTheme` as `__unstable` or `__experimental`. Apart from it being mobile only, we still haven't figured out a common design language between mobile/web so it might even change in the near future.

File: packages/components/src/mobile/dark-mode/index.native.js
Comment (koke) [inline packages/components/src/mobile/dark-mode/index.native.js]: Using `theme` for this could be misleading, as WordPress has already a "theme" concept for a different thing. I think we could borrow the term "preferred color scheme" from the web implementation of dark mode and implement this cross platform?

Just like we have a `useReducedMotion` hook, we could also have a `usePreferredColorScheme`. For now, since you can't use hooks in class components, we could still have a `withPreferredColorScheme` HOC for this.

The web implementation seems simple enough:

```jsx
const isDarkMode = useMediaQuery( '(prefers-color-scheme: dark)' );
```

File: packages/rich-text/src/component/index.native.js
Comment (Tug) [reply packages/rich-text/src/component/index.native.js]: I wouldn't port this HOC to the web, at least not in its current state. It uses an internal event emitter to listen to native events and refresh the styles and is very much tied to the use case of dark mode right now. I think we will want to refactor this even for native first and use the store to dispatch events at least.

File: packages/components/src/mobile/dark-mode/index.native.js
Comment (Tug) [reply packages/components/src/mobile/dark-mode/index.native.js]: Agreed, I much prefer this approach 👍

File: packages/rich-text/src/component/index.native.js
Comment (etoledom) [reply packages/rich-text/src/component/index.native.js]: I agree too. There already were some issues with linting.

File: packages/rich-text/src/component/index.native.js
Comment (gziolo) [reply packages/rich-text/src/component/index.native.js]: Sounds like `__experimentalWithDarkMode` then :)

File: packages/components/src/mobile/dark-mode/index.native.js
Comment (gziolo) [reply packages/components/src/mobile/dark-mode/index.native.js]: `withPreferredColorScheme` - sounds like a good one. We still might want to refactor the hook as follows:

```js
const isDarkMode = usePreferredColorScheme( 'dark' );
```
and use it inside the HOC. `useMediaQuery` might be not applicable to native mobile, right?

**Review by Tug — APPROVED** [review]
This LGTM 👍
Since I authored some good bits of it, I think it could benefit from another review. @etoledom do you mind having a look?
Given that `rnmobile/master` is merged in, it might be a bit tricky to review now, I think I can try to group all our changes in a single commit if that makes sense.

**Comment by Tug** [comment]
Ok this is rebased ontop of `master`, all commits related to fixing `master` for our RN build are grouped into 2 commits: the first one is your initial commit and the second one groups all my previous commits. The last commit is a squashed merge of `rnmobile/master`. Should be much easier to review now :)

File: .travis.yml
Comment (Tug) [reply .travis.yml]: We might actually need to branch off again as we focus on the Open Beta. Keeping it for now 👍

File: packages/block-editor/CHANGELOG.md
Comment (Tug) [inline packages/block-editor/CHANGELOG.md]: Not sure what this change is doing here 🤔

Comment (Tug) [reply packages/block-editor/CHANGELOG.md]: Looks like we're fine, this is part of https://github.com/WordPress/gutenberg/pull/17195/files

Comment (gziolo) [reply packages/block-editor/CHANGELOG.md]: Yes, this PR landed yesterday 👍

File: .travis.yml
Comment (gziolo) [reply .travis.yml]: I'm sure you will branch more often so it's fine to keep it here until we merge the mobile repo in here.

**Comment by etoledom** [comment]
> Since I authored some good bits of it, I think it could benefit from another review. @etoledom do you mind having a look?

Sure! I'll give it a deep test in both iOS and Android 👍

File: packages/api-fetch/src/index.native.js
Comment (mchowning) [inline packages/api-fetch/src/index.native.js]: Instead of `console.warn`, could we make this `console.info` or `console.log` so we don't add another yellow box warning to the debug apps?

Comment (Tug) [reply packages/api-fetch/src/index.native.js]: I think it's pretty alarming if the app is trying to make an api request and we should be aware of it. At least it's as alarming as state update warning or deprecation notice imo.
What we could do if you think those warnings are disrupting the experience when testing on mobile is disable them completely. If we want to see those we can simply attach a debugger.

Comment (Tug) [reply packages/api-fetch/src/index.native.js]: Disabled in https://github.com/wordpress-mobile/gutenberg-mobile/pull/1366/commits/fb6db571f796adef5eeaaa310e3d0c4ec88b1028

Comment (mchowning) [reply packages/api-fetch/src/index.native.js]: > I think it's pretty alarming if the app is trying to make an api request and we should be aware of it.

I'm seeing that warning every time I run the app. Is that expected?

**Comment by etoledom** [comment]
✅ (I created this PR so I can't approve it on GH ¯\\\_(ツ)\_/¯ )
I did an deep test on iOS and Android example apps, and also on WPiOS and WPAndroid.

I found some issues on the way, but by communicating early with @Tug , they have been fixed already 🎉 

- The Heading block Hx buttons were no displaying (fixed on https://github.com/WordPress/gutenberg/pull/17228/commits/74ee26ffc9e95e692ac6706c1ec65d2fb1d59810)
- The Post Title was being removed when the editor loaded (fixed on https://github.com/WordPress/gutenberg/pull/17228/commits/5a7ee6d58633d0ba2265dca97728f0a2ced61565)
- A warning about reaching the maximum number of listener reappeared (fixed on https://github.com/WordPress/gutenberg/pull/17228/commits/d62ead04984d4da2cfdb4b4ef2bcd4ff5a0d285a)

After the Hx buttons bug was fixed, there's one remaining issue related to these buttons:
- The Hx buttons on the Heading block won't change its icon and subscript color when it becomes active.

I'm happy to merge and fix this later on, or if you prefer to fix it right away that works for me too!
You can choose!

Great work @Tug as always, and thanks a lot for all this!

:shipit:  !

File: packages/api-fetch/src/index.native.js
Comment (Tug) [reply packages/api-fetch/src/index.native.js]: Yeah I see it as well, I haven't tracked down the issue yet, but we might have to when we tackle editing image options since it seems related to media.

Comment (mchowning) [reply packages/api-fetch/src/index.native.js]: Thanks for explaining. If this is a we-need-to-fix-this-as-soon-as-reasonably-possible type of thing, I have no concern about leaving the warning if you want. Alternatively, removing the warning and filing an issue to add the warning back in (and fix whatever is causing it to fire) could make sense so this doesn't get forgotten.

**Review by mchowning — APPROVED** [review]
Smoke tested this on both platforms and it LGTM! Great job @etoledom and @Tug ! 🙇 

The only issue I saw was the issue that @etoledom noted with the Heading level icons text color not updating appropriately when selected (i.e. they stay gray instead of turning white-ish).

Screen Shot 2019-09-27 at 9 57 52 AM

I would lean toward merging this PR as-is and fixing that minor issue in a follow-up PR myself.

**Comment by gziolo** [comment]
Nice work on this PR 👍

File: packages/components/src/icon-button/index.js
Comment (youknowriad) [inline packages/components/src/icon-button/index.js]: This line breaks the web because ariaPressed is not a valid react prop in the web. Dashicon addresses t his properly but not the Icon component.

Comment (etoledom) [reply packages/components/src/icon-button/index.js]: This PR removes this `ariaPressed` https://github.com/WordPress/gutenberg/pull/17356
It's practically ready to merge.

Comment (youknowriad) [reply packages/components/src/icon-button/index.js]: Cool, thank you

Comment (etoledom) [reply packages/components/src/icon-button/index.js]: Thanks for the heads up! I'll let you know when it gets merged.

**Comment by youknowriad** [comment]
I think changes to Icon component here also broke this https://github.com/WordPress/gutenberg/issues/17719 

Any idea how to solve it? it looks like previously the size passed to custom component icons was "undefined" and now it's "24"

**Pull request merged**
